src/celestia/qt/qtappwin.cpp

```diff
@@ -217,10 +217,11 @@ void CelestiaAppWindow::init(const CelestiaCommandLineOptions& options)
 
     if (celestia_data_dir.isEmpty()) {
         QString dataDir = CONFIG_DATA_DIR;
-        QString celestia_data_dir = dataDir;
-        QDir::setCurrent(celestia_data_dir);
+        QDir::setCurrent(dataDir);
+        m_dataHome = QDir::currentPath();
```

Comment:
This should be instead lines 219-221

Reply:
m_dataHome can be used to take CONFIG_DATA_DIR to the setCurrent function.  But you still need to set it with currentPath call afterward.  currentPath automatically creates an absolute path from the working directory.  If you only set m_dataHome from CONFIG_DATA_DIR when you use the --dir command line option with a relative path, there are conditions you get a URL passed to the browser trying to find the help guide that is a relative path.  The browser or qt will pop error messages - depending on the path you used.  The code requested changed in previous submission was used/run and did fail upon the --dir checkout test.

While the --dir command executes a different line in the submitted code, the fall back in the first few lines of this module will set the CONF_DATA_DIR to the relative path "./", so it will also fail in a similar manner.

The alternative is to use some combination of QDIR absolute path functions, but that function is not specific to what of multiple QDir paths it reads, though I suppose you could create the absolute path first then use it to set m_dataHome,

The other issue is canonical (spelling?) paths when a path contains a link.  I have tried links in several path iterations and this code has successfully resolved the link and sent it to the browser.  No documentation says it will, it just has upon trial.

Reply:
okay, let's use this way then